### PR TITLE
Feature/summary unpack order

### PR DESCRIPTION
**Pull request by ahartikainen**
Have the opportunity to unpack nD objects (n > 1) against either C-order (iterate last dimension first) or F-order (iterate first dimension first).

Implemented only for "wide" format.

### CODE REVIEW

**Comment by ahartikainen** [comment]
I still need to add test for the order.

File: arviz/tests/test_stats.py [hunk 1]
```diff
@@ -134,10 +139,19 @@ def test_summary_nan(centered_eight):
 
 
 def test_summary_bad_fmt(centered_eight):
+    with pytest.raises(TypeError):
+        summary(centered_eight, fmt=1)
     with pytest.raises(TypeError):
         summary(centered_eight, fmt="bad_fmt")
 
 
+def test_summary_bad_unpack_order(centered_eight):
```
Comment (canyon289) [inline arviz/tests/test_stats.py]: Can these be parameterized? That way they'll fail independently and we'll get an accurate test count

File: arviz/tests/test_stats.py [hunk 2]
```diff
@@ -134,10 +139,19 @@ def test_summary_nan(centered_eight):
 
 
 def test_summary_bad_fmt(centered_eight):
+    with pytest.raises(TypeError):
```
Comment (canyon289) [inline arviz/tests/test_stats.py]: Can these be parameterized? That way they'll fail independently and we'll get an accurate test count

File: arviz/stats/stats.py [hunk 3]
```diff
@@ -786,7 +795,9 @@ def summary(
             if len(values.shape[1:]):
                 metric = list(values.metric.values)
                 data_dict = {}
-                for idx in np.ndindex(values.shape[1:]):
+                for idx in np.ndindex(values.shape[1:] if order == "C" else values.shape[1:][::-1]):
```
Comment (canyon289) [inline arviz/stats/stats.py]: What does the if statement do here?

Comment (ahartikainen) [reply arviz/stats/stats.py]: Small numpy indexing trick to use 'F' order.

- C = iterate last dim first
- F = iterate first dim first

File: arviz/tests/test_stats.py (hunk 1 above)
Comment (ahartikainen) [reply arviz/tests/test_stats.py]: Yes

File: arviz/tests/test_stats.py (hunk 2 above)
Comment (ahartikainen) [reply arviz/tests/test_stats.py]: Yes

**Comment by ahartikainen** [comment]
I think this is good to go.

It is a small change, but really annoying to fix by hand.

**Comment by canyon289** [comment]
LGTM

File: arviz/stats/stats.py (hunk 3 above)
Comment (canyon289) [reply arviz/stats/stats.py]: But in this statement if `order =="C"` goes nowhere? Or in other words if `order == "C"` it doesn't look like there's any code that gets executed?

Comment (ahartikainen) [reply arviz/stats/stats.py]: Ah, conditional expression https://docs.python.org/3/reference/expressions.html#conditional-expressions

`values.shape[1:] if order == "C" else values.shape[1:][::-1]`

`value_bool_true if bool else value_bool_false`

you could also chain them

`value1 if bool else value2 if bool2 else value3 if ....`

**Pull request merged**